controller/state.go

```diff
@@ -664,17 +664,19 @@ func (m *appStateManager) CompareAppState(app *v1alpha1.Application, project *ap
 				Destination: app.Spec.Destination,
 				Sources:     sources,
 			},
-			Status:    syncCode,
-			Revisions: manifestRevisions,
+			Status:     syncCode,
+			Revisions:  manifestRevisions,
+			ComparedAt: &now,
```

Comment:
What about `status.reconciledAt`? This value will be updated by the controller whenever there are updates to the applications status, including changing the health state.

I am not sure this value provide much information. Based on the issue, this is to perform notifications. You can use this value to delay the notification, and use `oncePer: app.status.sync.revision` to only sent it once after the sync.